### DISCUSSION TITLE
bbworker:  Redis connection failed - undefined

### DISCUSSION THREAD

**Discussion by Bogey** [Budibase/budibase#2895] [Help] [answered]
After successfully running Budibase for quite a while via Portainer on a Synology NAS. However, I now cannot get it to work anymore.
The culprit seems to be the Redis service. bbworker (and bbapps) containers cannot connect to the Redis service.
I'm using the docker-compose file from budibase with a few minor adjustments: paths and container names.
Portainer has the option to load .env file, so it's the same as the hosting.properties file (of course with my passwords).

bbworker starts correct until redis comes in the picture:
> yarn run v1.22.5
> $ node src/index.js
> Worker running on {"address":"::","family":"IPv6","port":4003}
> Redis connection failed - Did not successfully connect in timeout

I don't have IPv6, perhaps that's a problem?

This is my docker-compose file:
[budibase portainer stack.txt](https://github.com/Budibase/budibase/files/7287117/budibase.portainer.stack.txt)

**Comment by mjashanks** [comment]
I'll have to get someone who knows more about this to answer. 

Did this happen after an update? Have you manually checked connectivity between the containers, and checked that Redis is definately running?

**Reply by Bogey** [reply]
Yes, it happened after an update.
All containers are running, Redis too.

**Comment by shogunpurple** [comment, marked as answer]
Hi @Bogey - can you try a few things just to debug:

Are you able to SSH into one of your containers, and successfully connect to the `redis-service` using `redis-cli`? Just want to make sure this isn't a networking thing first.

Failing that, please try and update your configs `REDIS_URL` to be the following:

`redis://default:yourredispassword@redis-service:6379`

then re-run `docker-compose --env-file hosting.properties up -d`

Let me know if any of this helps. We will be rolling out the latest to self hosters today, so there's a possibility your issue has been resolved in all the bug fixes that are coming in the latest release.

**Reply by Bogey** [reply]
Thanks Martin, changing REDIS_URL did the trick. I changed it (after testing your solution) eventually from:
`REDIS_URL: redis-service:6379`
to
`REDIS_URL: redis://default:${REDIS_PASSWORD}@redis-service:6379`

**Reply by melanki** [reply]
@shogunpurple @Bogey 
I am having the same issue, which files contains the REDIS_URL that needs to be updated?

**Reply by Bogey** [reply]
The docker-compose file. You can find it on Github at:
budibase/hosting/docker-compose.yaml

**Comment by karthikkangokar** [comment]
I am getting error redis connection failed . did not sucessfully connect during timeout after update to latest version
